hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenDissociateFromAccountHandler.java

```diff
@@ -167,9 +170,8 @@ public void handle(@NonNull final HandleContext context) {
         // get changes to account and token relations
         final var updatedTokenRels =
                 new TokenRelListCalculator(tokenRelStore).removeTokenRels(account, tokenRelsToRemove);
-        final var newHeadTokenId = updatedTokenRels.updatedHeadTokenId() != null
-                ? updatedTokenRels.updatedHeadTokenId()
-                : account.headTokenNumber();
+        final var newHeadTokenId =
+                updatedTokenRels.updatedHeadTokenId() != null ? updatedTokenRels.updatedHeadTokenId() : null;
```

Comment:
unnecessary null check, could be
`        final var newHeadTokenId = updatedTokenRels.updatedHeadTokenId();`